core/src/main/java/io/questdb/std/str/AbstractCharSequence.java

```diff
@@ -57,6 +57,14 @@ public final CharSequence subSequence(int start, int end) {
         if (start == 0 && end == length()) {
             return this;
         }
+        // it's tempting apply this optimization:
+        // if (start == end) {
```

Comment:
this is a wrong assumption - `start = end` is asking of one char and not empty string. `end` is inclusive (as your PR fixes :) )

Reply:
are you sure?
```
@Test
public void testSubSequence_startEqualsEndReturnsEmptyString() {
    // sanity test
    TestUtils.assertEquals("", "foo".subSequence(1, 1));

    // try with our own impl of AbstractCharSequence
    byte[] bytes = "foo".getBytes(StandardCharsets.UTF_16LE);
    DirectCharSequence seq = new DirectCharSequence();
    long buff = Unsafe.malloc(bytes.length, MemoryTag.NATIVE_DEFAULT);
    try {
        for (int i = 0; i < bytes.length; i++) {
            Unsafe.getUnsafe().putByte(buff + i, bytes[i]);
        }
        seq.of(buff, buff + bytes.length);
        TestUtils.assertEquals("foo", seq);
        TestUtils.assertEquals("", seq.subSequence(1, 1));
    } finally {
        Unsafe.free(buff, bytes.length, MemoryTag.NATIVE_DEFAULT);
    }
}
```

This PR is fixing the case where a `subSequence()` callers ask for `start=end`, and they are expected to get an empty string back. 

think of this query: `select '' from long_sequence(1);`. The token `''` has a length 2. And when you call `unquote()`, then you should get an empty string back. That's currently broken (for some implementations of `CharSequence`) and this PR is fixing it.

Reply:
You are right. I rushed to conclusion because `testEmptyQuotedString` failed with this change. It appears returning `""` highlighted another bug in expression parsing system. We dereference 0 char here on empty str:

![image](https://user-images.githubusercontent.com/7276403/205096118-0c2b26e5-9181-4642-868c-2a1a60dff490.png)

It only works because floating sequence doesn't check bounds :( This is something we need to fix. Returning `""` is legit.

Reply:
that's a very good finding! I'll fix this too. 
without knowing much of a context: the fix looks trivial - just a length check. and a test should be also relatively simple - at least when testing `ExpressionNode` in isolation.

Reply:
I add a fix for this very issue in ExpressionNode. However it seems there are other similar problem: `charAt(0)` is called without checking length first. I'll explore it further and if it's real then I will send a separate PR.

Reply:
an alternative would making `token` null when alias length becomes 0?

Reply:
I think an empty string is a valid literal and the ExpressionNode should handle it correctly. 
Otherwise, the contact would be: "a token can be null, token.length>0, but never token.length=0" and that sounds funky.

Reply:
ok,  cool, so with this fix we can return `""` from the sub-string as in your musings earlier. This will also test the fix